cmd/backupfilemigrator/file_migrator.go

```diff
@@ -0,0 +1,197 @@
+	err := jsonfast.Unmarshal([]byte(gjson.GetBytes(lineBytes, "event_payload.batch").String()), &gatewayBatchReq)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to unmarshall gateway events")
+	}
+	listOfNewEvents := make([]*newFileFormat, 0)
+	userId := gjson.GetBytes(lineBytes, "user_id").String()
+	for _, singleEvent := range gatewayBatchReq {
+		payloadBytes, err := json.Marshal(singleEvent)
+		if err != nil {
+			return nil, errors.Wrap(err, "unable to marshall single event")
+		}
+		j := &newFileFormat{}
+		j.UserID = userId
+		j.EventPayload = payloadBytes
+		j.CreatedAt = createdAt
+		j.MessageID = misc.GetStringifiedData(singleEvent["messageId"])
+		listOfNewEvents = append(listOfNewEvents, j)
+	}
+	return listOfNewEvents, nil
+}
+
+// uploadFile creates a new format file and upload to storage
+func (m *fileMigrator) uploadFile(ctx context.Context, jobs []*newFileFormat, sourceId, workspaceId, instanceId string) error {
+	firstJobCreatedAt := jobs[0].CreatedAt.UTC()
+	lastJobCreatedAt := jobs[len(jobs)-1].CreatedAt.UTC()
+
+	gzWriter := fileuploader.NewGzMultiFileWriter()
+	localFilePath := path2.Join(
+		lo.Must(misc.CreateTMPDIR()),
+		"rudder-backups",
+		sourceId,
+		fmt.Sprintf("%d_%d_%s.json.gz", firstJobCreatedAt.Unix(), lastJobCreatedAt.Unix(), workspaceId),
+	)
+
+	for _, job := range jobs {
+		jobBytes, err := json.Marshal(job)
+		if err != nil {
+			_ = gzWriter.Close()
+			return errors.Wrap(err, "failed to marshal job")
+		}
+		if _, err := gzWriter.Write(localFilePath, append(jobBytes, '\n')); err != nil {
+			_ = gzWriter.Close()
+			return errors.Wrap(err, "write to local file failed")
+		}
+	}
+	if err := gzWriter.Close(); err != nil {
+		return errors.Wrap(err, "failed to close writer")
+	}
+	defer func() { _ = os.Remove(localFilePath) }()
+
+	localFile, err := os.Open(localFilePath)
+	if err != nil {
+		return errors.Wrap(err, "failed to open local file")
+	}
+	defer func() { _ = localFile.Close() }()
+	prefixes := []string{
+		sourceId,
+		"gw",
+		firstJobCreatedAt.Format("2006-01-02"),
+		fmt.Sprintf("%d", firstJobCreatedAt.Hour()),
+		instanceId,
+	}
+	_, err = m.fileManager.Upload(ctx, localFile, prefixes...)
+	if err != nil {
+		return errors.Wrap(err, "failed to upload file")
+	}
+	return nil
+}
+
+// download file locally
+func (m *fileMigrator) downloadFile(ctx context.Context, filePath string) (string, error) {
+	filePathTokens := strings.Split(filePath, "/")
+	// backup file is present in folder named instance id
+	// e.g. rudder-saas/cars/cars-v0-rudderstack-10/gw_jobs_11796.317963152.317994396.1703547948443.1703548519552.2HajBvyyfPW3n0QNSaYH7eEXt1z.gz
+	var err error
+	dumpDownloadPathDirName := "/rudder-s3-dumps/"
+	tmpdirPath := strings.TrimSuffix(kitConfig.GetString("RUDDER_TMPDIR", "/tmp"), "/")
+	if tmpdirPath == "" {
+		tmpdirPath, err = os.UserHomeDir()
+		if err != nil {
+			return "", errors.Wrap(err, "failed to get user home directory")
+		}
+	}
+	path := fmt.Sprintf(
```

Comment:
Variable 'path' collides with imported package name 
```suggestion
	tempPath := fmt.Sprintf(
```